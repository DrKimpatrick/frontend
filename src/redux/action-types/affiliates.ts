export enum AffiliateTypes {
  AddMoreInfo = 'Affiliate/AddMoreInfo',
  Errors = 'Affiliate/Errors',
  Loading = 'Affiliate/Loading'
}

// export interface CompanyOrSchool {
//   _id?: string;
//   name: string;
//   address: string;
//   website: string;
//   accountManagerName: string;
//   email: string;
//   phone: string;
//   userId?: string;
//   createdAt?: string;
//   updatedAt?: string;
// }

interface AddMoreInfo {
  type: typeof AffiliateTypes.AddMoreInfo;
  payload: {
    data: any;
  };
}

interface Errors {
  type: typeof AffiliateTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof AffiliateTypes.Loading;
}

export type CompanyActionTypes = AddMoreInfo | Errors | Loading;
