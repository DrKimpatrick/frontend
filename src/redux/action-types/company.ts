export enum CompanyTypes {
  AddCompany = 'Company/AddCompany',
  GetSpecificCompany = 'Company/GetSpecificCompany',
  UpdateCompany = 'Company/UpdateCompany',
  DeleteCompany = 'Company/DeleteCompany',
  Errors = 'Company/Errors',
  Loading = 'Company/Loading'
}

export interface CompanyOrSchool {
  _id?: string;
  name: string;
  address: string;
  website: string;
  accountManagerName: string;
  email: string;
  phone: string;
  userId?: string;
  createdAt?: string;
  updatedAt?: string;
}

interface AddCompany {
  type: typeof CompanyTypes.AddCompany;
  payload: {
    data: CompanyOrSchool;
  };
}

interface Errors {
  type: typeof CompanyTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof CompanyTypes.Loading;
}

export type CompanyActionTypes = AddCompany | Errors | Loading;
