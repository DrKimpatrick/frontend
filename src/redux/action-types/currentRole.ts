export const CURRENT_ROLE_LOADING = 'CURRENT_ROLE_LOADING';
export const CURRENT_ROLE_FAIL = 'CURRENT_ROLE_FAIL';
export const CURRENT_ROLE_SUCCESS = 'FORGOT_SUCCESS';

export interface CurrentRole {
    email: string
}

export interface currentRoleLoading  {
    type: typeof CURRENT_ROLE_LOADING
}

export interface currentRoleFail  {
    type: typeof CURRENT_ROLE_FAIL,
    payload: string
}

export interface currentRoleSuccess  {
    type: typeof CURRENT_ROLE_SUCCESS,
    payload: any
}

export type CurrentRoleDispatchTypes = currentRoleLoading | currentRoleFail | currentRoleSuccess;
