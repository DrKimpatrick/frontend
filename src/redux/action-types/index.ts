// This is where we will add action types