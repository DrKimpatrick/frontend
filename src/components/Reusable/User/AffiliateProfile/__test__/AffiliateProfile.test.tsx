import React from 'react';
import ReactDOM from 'react-dom';
import Component from '../AffiliateProfile';
import { cleanup, render } from '@testing-library/react';
import { UserRole } from 'redux/action-types/user';

const props = {
  courses: [
    {
      currentLangSpecsUpdated: false,
      verificationStatus: 'pending',
      description: 'desc',
      _id: 'id',
      name: 'React Crash Course',
      instructor: 'Ryan Dahl',
      languageTaught: 'English',
      existingCourseLink: 'link',
      coverImageLink: 'link',
      level: 'advanced',
      duration: '5 Hr',
      userId: {
        _id: 'id',
        username: 'id',
        email: 'email'
      },
      createdAt: 'date',
      updatedAt: 'date'
    }
  ]
};
describe('AffiliateProfile', () => {
  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Component {...props} />, div);
  });

  it('should create snapshot', () => {
    const tree = render(<Component {...props} />);

    expect(tree).toMatchSnapshot();
  });
});
