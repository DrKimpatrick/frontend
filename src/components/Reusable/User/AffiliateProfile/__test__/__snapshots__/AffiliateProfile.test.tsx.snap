// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AffiliateProfile should create snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="affiliateProfile"
      >
        <div
          class="affiliateStatistic"
        >
          <ul
            class="listOfStatistic"
          >
            <li>
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5 16c0 3.87 3.13 7 7 7s7-3.13 7-7v-4H5v4zM16.12 4.37l2.1-2.1-.82-.83-2.3 2.31C14.16 3.28 13.12 3 12 3s-2.16.28-3.09.75L6.6 1.44l-.82.83 2.1 2.1C6.14 5.64 5 7.68 5 10v1h14v-1c0-2.32-1.14-4.36-2.88-5.63zM9 9c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
                  />
                </svg>
                <span
                  class="name"
                >
                  Total clicks
                </span>
              </div>
              <div
                class="rate"
              >
                <span>
                  20%
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 9h4V5H3v4zm0 5h4v-4H3v4zm5 0h4v-4H8v4zm5 0h4v-4h-4v4zM8 9h4V5H8v4zm5-4v4h4V5h-4zm5 9h4v-4h-4v4zM3 19h4v-4H3v4zm5 0h4v-4H8v4zm5 0h4v-4h-4v4zm5 0h4v-4h-4v4zm0-14v4h4V5h-4z"
                  />
                </svg>
                <span
                  class="name"
                >
                  Conversation Rate
                </span>
              </div>
              <div
                class="rate"
              >
                <span>
                  20%
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 12c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zM6 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12-8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm-4 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="name"
                >
                  Conversation
                </span>
              </div>
              <div
                class="rate"
              >
                <span>
                  20%
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
                  />
                </svg>
                <span
                  class="name"
                >
                  Quarterly Commission
                </span>
              </div>
              <div
                class="rate"
              >
                <span>
                  20%
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.6 6.62c-1.44 0-2.8.56-3.77 1.53L12 10.66 10.48 12h.01L7.8 14.39c-.64.64-1.49.99-2.4.99-1.87 0-3.39-1.51-3.39-3.38S3.53 8.62 5.4 8.62c.91 0 1.76.35 2.44 1.03l1.13 1 1.51-1.34L9.22 8.2C8.2 7.18 6.84 6.62 5.4 6.62 2.42 6.62 0 9.04 0 12s2.42 5.38 5.4 5.38c1.44 0 2.8-.56 3.77-1.53l2.83-2.5.01.01L13.52 12h-.01l2.69-2.39c.64-.64 1.49-.99 2.4-.99 1.87 0 3.39 1.51 3.39 3.38s-1.52 3.38-3.39 3.38c-.9 0-1.76-.35-2.44-1.03l-1.14-1.01-1.51 1.34 1.27 1.12c1.02 1.01 2.37 1.57 3.82 1.57 2.98 0 5.4-2.41 5.4-5.38s-2.42-5.37-5.4-5.37z"
                  />
                </svg>
                <span
                  class="name"
                >
                  Total Links
                </span>
              </div>
              <div
                class="rate"
              >
                <span>
                  20%
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
                  />
                </svg>
                <span
                  class="name"
                >
                  New Links
                </span>
              </div>
              <div
                class="rate"
              >
                <span>
                  20%
                </span>
              </div>
            </li>
          </ul>
          <ul
            class="listOfStatistic"
          >
            <li>
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.5 13c-1.2 0-3.07.34-4.5 1-1.43-.67-3.3-1-4.5-1C5.33 13 1 14.08 1 16.25V19h22v-2.75c0-2.17-4.33-3.25-6.5-3.25zm-4 4.5h-10v-1.25c0-.54 2.56-1.75 5-1.75s5 1.21 5 1.75v1.25zm9 0H14v-1.25c0-.46-.2-.86-.52-1.22.88-.3 1.96-.53 3.02-.53 2.44 0 5 1.21 5 1.75v1.25zM7.5 12c1.93 0 3.5-1.57 3.5-3.5S9.43 5 7.5 5 4 6.57 4 8.5 5.57 12 7.5 12zm0-5.5c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zm9 5.5c1.93 0 3.5-1.57 3.5-3.5S18.43 5 16.5 5 13 6.57 13 8.5s1.57 3.5 3.5 3.5zm0-5.5c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2z"
                  />
                </svg>
                <span
                  class="name"
                >
                  Converted
                </span>
              </div>
              <div
                class="rate"
              >
                <span>
                  20%
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
                  />
                </svg>
                <span
                  class="name"
                >
                  Amount
                </span>
              </div>
              <div
                class="rate"
              >
                <span>
                  20%
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
                  />
                </svg>
                <span
                  class="name"
                >
                  Amount Paid
                </span>
              </div>
              <div
                class="rate"
              >
                <span>
                  20%
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
                  />
                </svg>
                <span
                  class="name"
                >
                  Balance
                </span>
              </div>
              <div
                class="rate"
              >
                <span>
                  20%
                </span>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="affiliateCourse"
        >
          <span
            class="title"
          >
            Courses
          </span>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="my-1"
              >
                <div
                  class="listCourseByLevel"
                >
                  <div
                    class="level w-full"
                  >
                    <span>
                      beginner
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="my-1"
              >
                <div
                  class="listCourseByLevel"
                >
                  <div
                    class="level w-full"
                  >
                    <span>
                      intermediate
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="my-1"
              >
                <div
                  class="listCourseByLevel"
                >
                  <div
                    class="level w-full"
                  >
                    <span>
                      advanced
                    </span>
                  </div>
                  <ul
                    class="bg-card-preview"
                  >
                    <li
                      class="listCourse"
                    >
                      <span
                        class="name"
                      >
                        React Crash Course
                      </span>
                      <ul
                        class="action"
                      >
                        <li>
                          <button
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M13 2.05v3.03c3.39.49 6 3.39 6 6.92 0 .9-.18 1.75-.48 2.54l2.6 1.53c.56-1.24.88-2.62.88-4.07 0-5.18-3.95-9.45-9-9.95zM12 19c-3.87 0-7-3.13-7-7 0-3.53 2.61-6.43 6-6.92V2.05c-5.06.5-9 4.76-9 9.95 0 5.52 4.47 10 9.99 10 3.31 0 6.24-1.61 8.06-4.09l-2.6-1.53C16.17 17.98 14.21 19 12 19z"
                              />
                            </svg>
                          </button>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="affiliateProfile"
    >
      <div
        class="affiliateStatistic"
      >
        <ul
          class="listOfStatistic"
        >
          <li>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 16c0 3.87 3.13 7 7 7s7-3.13 7-7v-4H5v4zM16.12 4.37l2.1-2.1-.82-.83-2.3 2.31C14.16 3.28 13.12 3 12 3s-2.16.28-3.09.75L6.6 1.44l-.82.83 2.1 2.1C6.14 5.64 5 7.68 5 10v1h14v-1c0-2.32-1.14-4.36-2.88-5.63zM9 9c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
                />
              </svg>
              <span
                class="name"
              >
                Total clicks
              </span>
            </div>
            <div
              class="rate"
            >
              <span>
                20%
              </span>
            </div>
          </li>
          <li>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 9h4V5H3v4zm0 5h4v-4H3v4zm5 0h4v-4H8v4zm5 0h4v-4h-4v4zM8 9h4V5H8v4zm5-4v4h4V5h-4zm5 9h4v-4h-4v4zM3 19h4v-4H3v4zm5 0h4v-4H8v4zm5 0h4v-4h-4v4zm5 0h4v-4h-4v4zm0-14v4h4V5h-4z"
                />
              </svg>
              <span
                class="name"
              >
                Conversation Rate
              </span>
            </div>
            <div
              class="rate"
            >
              <span>
                20%
              </span>
            </div>
          </li>
          <li>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 12c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zM6 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12-8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm-4 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <span
                class="name"
              >
                Conversation
              </span>
            </div>
            <div
              class="rate"
            >
              <span>
                20%
              </span>
            </div>
          </li>
          <li>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
                />
              </svg>
              <span
                class="name"
              >
                Quarterly Commission
              </span>
            </div>
            <div
              class="rate"
            >
              <span>
                20%
              </span>
            </div>
          </li>
          <li>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.6 6.62c-1.44 0-2.8.56-3.77 1.53L12 10.66 10.48 12h.01L7.8 14.39c-.64.64-1.49.99-2.4.99-1.87 0-3.39-1.51-3.39-3.38S3.53 8.62 5.4 8.62c.91 0 1.76.35 2.44 1.03l1.13 1 1.51-1.34L9.22 8.2C8.2 7.18 6.84 6.62 5.4 6.62 2.42 6.62 0 9.04 0 12s2.42 5.38 5.4 5.38c1.44 0 2.8-.56 3.77-1.53l2.83-2.5.01.01L13.52 12h-.01l2.69-2.39c.64-.64 1.49-.99 2.4-.99 1.87 0 3.39 1.51 3.39 3.38s-1.52 3.38-3.39 3.38c-.9 0-1.76-.35-2.44-1.03l-1.14-1.01-1.51 1.34 1.27 1.12c1.02 1.01 2.37 1.57 3.82 1.57 2.98 0 5.4-2.41 5.4-5.38s-2.42-5.37-5.4-5.37z"
                />
              </svg>
              <span
                class="name"
              >
                Total Links
              </span>
            </div>
            <div
              class="rate"
            >
              <span>
                20%
              </span>
            </div>
          </li>
          <li>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
                />
              </svg>
              <span
                class="name"
              >
                New Links
              </span>
            </div>
            <div
              class="rate"
            >
              <span>
                20%
              </span>
            </div>
          </li>
        </ul>
        <ul
          class="listOfStatistic"
        >
          <li>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.5 13c-1.2 0-3.07.34-4.5 1-1.43-.67-3.3-1-4.5-1C5.33 13 1 14.08 1 16.25V19h22v-2.75c0-2.17-4.33-3.25-6.5-3.25zm-4 4.5h-10v-1.25c0-.54 2.56-1.75 5-1.75s5 1.21 5 1.75v1.25zm9 0H14v-1.25c0-.46-.2-.86-.52-1.22.88-.3 1.96-.53 3.02-.53 2.44 0 5 1.21 5 1.75v1.25zM7.5 12c1.93 0 3.5-1.57 3.5-3.5S9.43 5 7.5 5 4 6.57 4 8.5 5.57 12 7.5 12zm0-5.5c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zm9 5.5c1.93 0 3.5-1.57 3.5-3.5S18.43 5 16.5 5 13 6.57 13 8.5s1.57 3.5 3.5 3.5zm0-5.5c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2z"
                />
              </svg>
              <span
                class="name"
              >
                Converted
              </span>
            </div>
            <div
              class="rate"
            >
              <span>
                20%
              </span>
            </div>
          </li>
          <li>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
                />
              </svg>
              <span
                class="name"
              >
                Amount
              </span>
            </div>
            <div
              class="rate"
            >
              <span>
                20%
              </span>
            </div>
          </li>
          <li>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
                />
              </svg>
              <span
                class="name"
              >
                Amount Paid
              </span>
            </div>
            <div
              class="rate"
            >
              <span>
                20%
              </span>
            </div>
          </li>
          <li>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
                />
              </svg>
              <span
                class="name"
              >
                Balance
              </span>
            </div>
            <div
              class="rate"
            >
              <span>
                20%
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="affiliateCourse"
      >
        <span
          class="title"
        >
          Courses
        </span>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              class="my-1"
            >
              <div
                class="listCourseByLevel"
              >
                <div
                  class="level w-full"
                >
                  <span>
                    beginner
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              class="my-1"
            >
              <div
                class="listCourseByLevel"
              >
                <div
                  class="level w-full"
                >
                  <span>
                    intermediate
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              class="my-1"
            >
              <div
                class="listCourseByLevel"
              >
                <div
                  class="level w-full"
                >
                  <span>
                    advanced
                  </span>
                </div>
                <ul
                  class="bg-card-preview"
                >
                  <li
                    class="listCourse"
                  >
                    <span
                      class="name"
                    >
                      React Crash Course
                    </span>
                    <ul
                      class="action"
                    >
                      <li>
                        <button
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M13 2.05v3.03c3.39.49 6 3.39 6 6.92 0 .9-.18 1.75-.48 2.54l2.6 1.53c.56-1.24.88-2.62.88-4.07 0-5.18-3.95-9.45-9-9.95zM12 19c-3.87 0-7-3.13-7-7 0-3.53 2.61-6.43 6-6.92V2.05c-5.06.5-9 4.76-9 9.95 0 5.52 4.47 10 9.99 10 3.31 0 6.24-1.61 8.06-4.09l-2.6-1.53C16.17 17.98 14.21 19 12 19z"
                            />
                          </svg>
                        </button>
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
